SU-HSDO/suhumsci
# STN-560 : Dark Color Inversion Toggle & Secondary Nav + Main Nav Dropdown variant styles

# READY FOR REVIEW

## Summary
The work in this PR adds a dark color inversion theme setting to the Colorful and Traditional themes. This theme setting adds the class of `hb-dark-inversion` to the `<html>` tag which is used to target specific blocks and views on a page so that they can have dark inversion styles applied to them.

The following items have been styled to display their dark color inversion when the dark color inversion theme setting is activated:
- [x] Sidebar (secondary) navigation
- [x] Special menus
- [x] Hover state of items located within the main navigation item dropdowns

<img width="1413" alt="dark inversion" src="https://user-images.githubusercontent.com/12678977/97194085-1e81ae00-1780-11eb-84c7-e994bd1c5f15.png">

## Need Review By (Date)
10/27/2020

## Urgency
medium

## Steps to Test
1. In your CLI, run `npm run test`. Confirm our tests pass.
2. To test this PR you will need to go to your theme settings and turn on (check) the `dark inversion variant` theme setting. This setting is available for both Colorful and Traditional. You will need to test both themes.
3. Verify the class of `hb-dark-inversion` has been added to the `<html>` tag when the theme setting is toggled on.
4. Review the following items display dark inversion variant styles:
    - [x] Sidebar (secondary) navigation
    - [x] Special menus
    * You can sync with staging to pull in the Special menu on https://sparkbox-sandbox-stage.stanford.edu/qa-text-area or your can add a special menu to one of your QA pages.
    - [x] Hover state of items located within the main navigation item dropdowns
5. I recommend also testing the dark inversion variant styles with and without the local masthead dark variant styles turned on and off.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
